### PR TITLE
Revert "Revert "partially supported SQL cloudcare API""

### PR DESCRIPTION
**Pull request by gcapalbo**
http://manage.dimagi.com/default.asp?239090
@snopoke everything looks ok to me except for the failing hard when footprint=True

Reverts dimagi/commcare-hq#13605

### CODE REVIEW

**Comment by snopoke** [comment]
@NoahCarnahan @emord I'm at DSA summit from tomorrow. Could one of you look at this please. I think we should be able to use the [get_dependent_case_info](https://github.com/dimagi/commcare-hq/blob/3fc9712c88d55212b35512b78675194459dd85e5/corehq/ex-submodules/casexml/apps/phone/cleanliness.py#L205) function to get the list of case_ids.

File: corehq/apps/cloudcare/api.py [hunk 1]
```diff
@@ -143,16 +150,16 @@ def _filter(res):
                             case_list,
                             self.domain,
                             strip_history=self.strip_history,
-                        ).values()
+                        )
```
Comment (dimagimon) [inline corehq/apps/cloudcare/api.py]: E126 continuation line over-indented for hanging indent

**Comment by emord** [comment]
@gcapalbo @snopoke https://github.com/dimagi/commcare-hq/pull/13606/commits/0bf69563d2f22d5a8bf0c8895d44a1425485b4aa

It's not clear to me whether this is the correct solution, but it seems to be a step in the right direction 

https://github.com/dimagi/commcare-hq/blob/revert-13605-revert-13597-sk/sql-cloudcare-api/corehq/apps/cloudcare/tests/test_api.py#L188-L209 do not pass on SQL domains. Are they also required?

**Comment by snopoke** [comment]
reworked it a bit and now all tests pass on both except `testGetAllStripHistory` since SQL doesn't support strip history (but that seems like just an optimization).

**Comment by gcapalbo** [comment]
@snopoke thanks, looks good. i'm just going to run a couple tests locally now before merging

**Comment by gcapalbo** [comment]
(local testing passed)

**Pull request merged**